tests/queries/0_stateless/01639_distributed_sync_insert_zero_rows.sql

```diff
@@ -0,0 +1,24 @@
+DROP TABLE IF EXISTS local;
+DROP TABLE IF EXISTS distributed;
+
+CREATE TABLE local (x UInt8) ENGINE = Memory;
+CREATE TABLE distributed AS local ENGINE = Distributed(test_cluster_two_shards, currentDatabase(), local, x);
+
+SET insert_distributed_sync = 1;
+
+INSERT INTO distributed SELECT number FROM numbers(256) WHERE number % 2 = 0;
+SELECT count() FROM local;
+SELECT count() FROM distributed;
+
+TRUNCATE TABLE local;
+INSERT INTO distributed SELECT number FROM numbers(256) WHERE number % 2 = 1;
+SELECT count() FROM local;
+SELECT count() FROM distributed;
+
+TRUNCATE TABLE local;
+INSERT INTO distributed SELECT number FROM numbers(256) WHERE number < 128;
+SELECT count() FROM local;
+SELECT count() FROM distributed;
+
+DROP TABLE local;
+DROP TABLE distributed;
```

Comment:
But how does this test ensures that blocks with zero rows will not be sent? It seems that it does not?

Reply:
This test does not check it.
It is checked manually with logs.

Reply:
It is only needed to ensure that neither of these cases breaks.